### PR TITLE
4.37.2 update

### PR DESCRIPTION
**Pull request by psteyer**
Updating Transformers to version 4.37.0 due to being effected by CVE-2023-7018 and Curate CVE-2023-6730 which both show `transformers>=4.36.0` as cleared.

Jira: https://anaconda.atlassian.net/browse/PKG-3827

`transformers` required two additional dependency updates:
 [`huggingface_hub`](https://github.com/AnacondaRecipes/huggingface_hub-feedstock/pull/6) -> [`transformers`](https://github.com/AnacondaRecipes/transformers-feedstock/pull/10)
[`tokenizer`](https://github.com/AnacondaRecipes/tokenizers-feedstock/pull/9) -> [`transformers`](https://github.com/AnacondaRecipes/transformers-feedstock/pull/10)
[`safetensor`](https://github.com/AnacondaRecipes/safetensors-feedstock/pull/3) -> [`transformers`](https://github.com/AnacondaRecipes/transformers-feedstock/pull/10)

### Changes
- version number
- sha
- `tokenizer` pinnings

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {% set name = "transformers" %}
-{% set version = "4.32.1" %}
+{% set version = "4.37.0" %}
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: The latest is `4.37.2`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 1edc8ae1de357d97c3d36b04412aa63d55e6fc0c4b39b419a7d380ed947d2252
+  sha256: 5a0fdee36168f751770f7036ce7a8787be14f8b0d8f29806c493b6cb819c6c83
 
 build:
   skip: True  # [py<37]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
  skip: True  # [py<38]
```
https://github.com/huggingface/transformers/blob/v4.37.2/setup.py#L447

File: recipe/meta.yaml [hunk 3]
```diff
@@ -37,7 +37,7 @@ requirements:
     - regex !=2019.12.17
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:37]:
```yaml
run:
  - python
  - datasets !=2.5.0 (We should try to remove it and see what breaks)
  - filelock
  - huggingface_hub >=0.19.3,<1
  - safetensors >=0.4.1
  - tokenizers >=0.14,<0.19
  - tqdm >=4.27
```

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:37]: `datasets` is required https://github.com/huggingface/transformers/blob/v4.37.2/utils/update_metadata.py#L39.

[numpy](https://github.com/huggingface/transformers/blob/v4.37.2/setup.py#L134), [packaging](https://github.com/huggingface/transformers/blob/v4.37.2/setup.py#L141), [pyyaml](https://github.com/huggingface/transformers/blob/v4.37.2/setup.py#L146), [regex](https://github.com/huggingface/transformers/blob/v4.37.2/setup.py#L153), and [requests](https://github.com/huggingface/transformers/blob/v4.37.2/setup.py#L154) too, see https://github.com/huggingface/transformers/blob/v4.37.2/setup.py#L416-L426

**Comment by psteyer** [comment]
Checking on the new failure: looks like it's missing a newer version of `0.4.2` for `safetensor`. We currently have `0.4.0`.

**Comment by JeanChristopheMorinPerso** [comment]
Yeah, it needs `safetensors >=0.4.1`

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:37]: @psteyer Pinnings should be updated for:
```
  - huggingface_hub >=0.19.3,<1
  - safetensors >=0.4.1
```

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**